core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -245,7 +245,17 @@ object TopicCommand extends Logging {
 
         newTopic.configs(configsMap)
         val createResult = adminClient.createTopics(Collections.singleton(newTopic))
-        createResult.all().get()
+        try {
+          createResult.all().get()
+        } catch {
+          case e: ExecutionException => {
+            val cause = e.getCause
+            if (cause.isInstanceOf[TopicExistsException] || topic.ifTopicDoesntExist()) {
```

Comment:
Shouldn't we re-throw all exceptions except `TopicExistsException` if `topic.ifTopicDoesntExist`?